lib/encode.js

```diff
@@ -0,0 +1,142 @@
+ * @param {Map} [options.appContextMap] - A map of JSON-LD Context URLs and
+ *   their encoded CBOR-LD values (must be values greater than 32767 (0x7FFF)).
+ * @param {Map} [options.appTermMap] - A map of JSON-LD terms and
+ *   their associated CBOR-LD term codecs.
+ * @param {diagnosticFunction} [options.diagnose] - A function that, if
+ * provided, is called with diagnostic information.
+ *
+ * @returns {Uint8Array} - The encoded CBOR-LD bytes.
+ */
+export async function toCborld({
+  jsonldDocument, documentLoader, appContextMap, appTermMap, diagnose}) {
+
+  let termEncodingMap = undefined;
+  appTermMap = appTermMap || new Map();
+
+  // get the term encoding map by processing the JSON-LD Contexts
+  try {
+    const contextUrls = getJsonldContextUrls({jsonldDocument});
+    termEncodingMap = await getTermCodecMap(
+      {contextUrls, appTermMap, documentLoader, encode: true});
+  } catch(e) {
+    // quietly ignore embedded context errors, generate uncompressed CBOR-LD
+    if(e.value !== 'ERR_EMBEDDED_JSONLD_CONTEXT_DETECTED') {
+      throw e;
+    }
+  }
+
+  // generate the encoded CBOR Map
+  const encodingMap = _generateEncodingMap({
+    jsonldDocument, appContextMap, appTermMap, termEncodingMap});
+
+  // determine the appropriate codec
+  const cborldObject = (termEncodingMap) ?
+    new CompressedCborldCodec() :
+    new UncompressedCborldCodec();
+  cborldObject.set({value: encodingMap});
+
+  // encode as CBOR
+  const cborldBytes = cbor.encode(cborldObject);
+
+  if(diagnose) {
+    diagnose('CBOR Encoding Map:');
+    diagnose(await cbor.Diagnose.diagnose(cborldBytes));
+    diagnose('CBOR Diagnostic Mode:');
+    diagnose(await cbor.Commented.comment(cborldBytes, {max_depth: 10}));
+  }
+
+  return cborldBytes;
+}
+
+// generate a CBOR encoding map given a JSON-LD document and term codec map
+function _generateEncodingMap(
+  {jsonldDocument, appContextMap, appTermMap, termEncodingMap}) {
+  const encodingMap = new Map();
+
+  // add the appTermMap under @codec if it exists
+  if(appTermMap && appTermMap.size > 0 && termEncodingMap) {
+    const codecValue = new CodecMapCodec();
+    codecValue.set({value: appTermMap, termCodecMap: termEncodingMap});
+    encodingMap.set(termEncodingMap.get('@codec').value, codecValue);
+  }
+
+  for(const [key, value] of Object.entries(jsonldDocument)) {
+    // check for undefined term keys
+    if(termEncodingMap && termEncodingMap.get(key) === undefined) {
+      throw new CborldError(
+        'ERR_UNKNOWN_CBORLD_TERM',
+        `Unknown term '${key}' was detected in JSON-LD input.`);
+    }
+
+    // get the encoded term
+    const term = (termEncodingMap) ?
+      termEncodingMap.get(key).value : key;
+
+    // set the encoded value
+    let encodedValue = value;
+    if(typeof value === 'object' && value.constructor === Object) {
+      encodedValue = _generateEncodingMap({
+        jsonldDocument: value, appContextMap, termEncodingMap});
```

Comment:
Either drop both curly braces together or break after first curly brace and before last curly brace; the current code here mixes both styles.

```suggestion
      encodedValue = _generateEncodingMap(
        {jsonldDocument: value, appContextMap, termEncodingMap});
```